zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/Collection.java

```diff
@@ -424,24 +425,28 @@ public static Collection update(
 
     private static Set<String> setViewerTeams(CollectionDescription desc, Zebedee zebedee, Session session)
             throws IOException, ZebedeeException {
-        Set<String> teamIds = new HashSet<>();
-        List<String> teamNames = desc.getTeams();
+        Set<String> teamIds;
 
-        if (teamNames == null) {
-            teamNames = new ArrayList<>();
-        }
+        if (cmsFeatureFlags().isJwtSessionsEnabled()) {
+            teamIds = new HashSet<>(desc.getTeams());
+        } else {
+            teamIds = new HashSet<>();
+            List<String> teamNames = desc.getTeams();
 
-        // TODO: Remove the following transitional code once Florence is updated to send the team IDs rather than the team names.
-        TeamsService teams = zebedee.getTeamsService();
-        for (String teamName : teamNames) {
-            Team team = teams.findTeam(teamName);
-            if (team == null) {
-                throw new NotFoundException("team assigned to collection expected but does not exist");
+            if (teamNames == null) {
+                teamNames = new ArrayList<>();
             }
 
-            teamIds.add(team.getId());
+            TeamsService teams = zebedee.getTeamsService();
+            for (String teamName : teamNames) {
+                Team team = teams.findTeam(teamName);
+                if (team == null) {
+                    throw new NotFoundException("team assigned to collection expected but does not exist");
```

Comment:
Need to be careful here as the team could be deleted before the collection is processed and the description updated. If that happens you will get this exception and the action you are trying to perform will fail.
I think it would be safer so quietly error here rather than block the action completing.